// Copyright 2018 The Hypebot Authors. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package hypebot.riot.v3;

service SummonerService {
  rpc GetSummoner(GetSummonerRequest) returns (Summoner) {}
}

message GetSummonerRequest {
  oneof key {
    int64 id = 1;
    int64 account_id = 2;
    string name = 3;
  }
}

message Summoner {
  int64 id = 1;
  int64 account_id = 2;
  string name = 3;

  int64 revision_date = 4;

  int64 summoner_level = 5;
  int32 profile_icon_id = 6;
}
